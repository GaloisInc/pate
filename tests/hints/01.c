int g1;
double g2;

int f1(int *i) {
  return *i + 1;
}

int main() {
  f1(&g1);
  return (int)((void*)&g1 - (void*)&g2);
}
