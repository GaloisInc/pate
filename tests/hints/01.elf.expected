VerificationHints { blockMappings = []
                  , functionEntries = [ ("f1", FunctionDescriptor { functionSymbol = "f1", functionAddress = 0x1000, functionArguments = ["i"] })
                                      , ("main", FunctionDescriptor { functionSymbol = "main", functionAddress = 0x1017, functionArguments = [] })
                                      ]
                  , dataSymbols = [ ("g1", SymbolExtent { symbolLocation = 0x4000, symbolSize = 4} )
                                  , ("g2", SymbolExtent { symbolLocation = 0x4008, symbolSize = 8} )
                                  ]
                  }
